In [1]:
import pandas as pd
import numpy as np
import statistics
df=pd.read_csv('Dataset_w6.xlsx - Data.csv')
df

country             Region  \
0                           Afghanistan       SouthernAsia   
1                               Albania     SouthernEurope   
2                               Algeria     NorthernAfrica   
3                                Angola       MiddleAfrica   
4                   Antigua and Barbuda          Caribbean   
..                                  ...                ...   
191  Venezuela (Bolivarian Republic of)       SouthAmerica   
192                            Viet Nam  South-easternAsia   
193                               Yemen        WesternAsia   
194                              Zambia      EasternAfrica   
195                            Zimbabwe      EasternAfrica   

     Surface area (km2)  Population in thousands (2017)  \
0                652864                           35530   
1                 28748                            2930   
2               2381741                           41318   
3               1246700                           29784   
4                   442                             102   
..                  ...                             ...   
191              912050                           31977   
192              330967                           95541   
193              527968                           28250   
194              752612                           17094   
195              390757                           16530   

     GDP per capita (current US$)  Agricultural production index   \
0                           623.2                             125   
1                          3984.2                             134   
2                          4154.1                             160   
3                          4714.1                             175   
4                         14764.5                              88   
..                            ...                             ...   
191                       11068.9                             118   
192                        2067.9                             136   
193                        1106.4                             136   
194                        1311.1                             179   
195                         890.4                              99   

     International trade: Imports (million US$)  
0                                          3568  
1                                          4669  
2                                         47091  
3                                          8790  
4                                           491  
..                                          ...  
191                                       16324  
192                                      174111  
193                                        6861  
194                                        7442  
195                                        5212  

[196 rows x 7 columns]

# 1 Find out how many countries are available in the dataset

In [24]:
l=len(df['country'].unique())
l

196

INSIGHT:
    Total number of coutries is 196.

# 3 Find out how many countries with respect to the region are available in the dataset

In [17]:
df.Region.value_counts(sort=True).to_frame()

Region
WesternAsia            18
EasternAfrica          17
Caribbean              16
WesternAfrica          15
SouthernEurope         12
SouthAmerica           12
South-easternAsia      11
EasternEurope          10
NorthernEurope         10
MiddleAfrica            9
SouthernAsia            9
CentralAmerica          8
WesternEurope           7
EasternAsia             7
NorthernAfrica          5
Polynesia               5
Melanesia               5
CentralAsia             5
SouthernAfrica          5
NorthernAmerica         4
Micronesia              4
Oceania                 2

# 4. On average which region has the highest
a. GDP per capita (current US$)
b.International trade: Imports (million US$)



In [19]:
df[['Region','GDP per capita (current US$)']].groupby('Region').mean().sort_values('GDP per capita (current US$)',ascending=False).head()

GDP per capita (current US$)
Region                                       
NorthernAmerica                  57659.000000
WesternEurope                    55387.128571
Oceania                          44823.250000
NorthernEurope                   42098.220000
EasternAsia                      27967.614286

In [21]:
df[['Region','International trade: Imports (million US$)']].groupby('Region').mean().sort_values('International trade: Imports (million US$)',ascending=False).head()

International trade: Imports (million US$)
Region                                                     
NorthernAmerica                               663552.250000
EasternAsia                                   451995.000000
WesternEurope                                 407498.428571
NorthernEurope                                113407.200000
Oceania                                       112914.500000

INSIGHTS:
On average Northern America has the highest
GDP per capita  and
International trade followed by Eastern Asia and Western Europe


# 5. Find out which region has more consistent with respect to GDP per capita (current US$)

In [22]:
df[['Region','GDP per capita (current US$)']].mode()

Region  GDP per capita (current US$)
0  WesternAsia                        1106.4

 Western Asia is more consistent with respect to GDP per capita .